### ISSUE TITLE
It does not work with nested modules

### ISSUE THREAD

**Issue by agrberg** [splitwise/cacheable#5]
For example it will fail if used inside `Foo::Bar::BazController`. Detailed error report and stack trace to come.

**Comment by agrberg** [comment]
```ruby
module Foo
  class Bar
    include Cacheable
  
    cacheable :foo
    
    def foo
      sleep 1
      10
    end
  end
end
```

Will generate the following as soon as it is loaded `NameError (wrong constant name Foo::BarCacher)`

Source of the error is how `Class#name` works and is used by [`Cacheable::MethodGenerator. method_interceptor_module_name`](https://github.com/splitwise/cacheable/blob/1f6ccbb5f33f0d796279a6f14e898e94a0163bed/lib/cacheable/method_generator.rb#L15).

**Comment by rofreg** [comment]
Closed by #8.
